static/src/views/Chat.vue

```diff
@@ -319,14 +348,23 @@ export default {
     _processMessages(messages) {
       const formattedMessages = [];
       messages.forEach((m) => {
+        const user = this.userDetails[m.sender_id];
+        let displayName = m.sender_name+'abc';
+        if(user){
+          displayName = user.study_id ? user.study_id + ": " : "";
+          displayName += user.username ? user.username : user.first_name;
+          displayName += ' ['+m.sender_name+']';
```

Comment:
could the code that generates a display name be pulled out to a function? Or is it different from the above (I didn't review super closely)

Reply:
It's close. I think it's a 50/50 situation since the second time around it's slightly different. I went for it though